content/src/main/content/jcr_root/apps/cq/gui/components/authoring/dialogsitewide/dialogsitewide.jsp

```diff
@@ -0,0 +1,17 @@
+<%@ page import="javax.jcr.Session" %>
+<%@ page import="com.day.cq.commons.jcr.JcrUtil" %>
+<%@page session="false"%>
+<%@taglib prefix="sling" uri="http://sling.apache.org/taglibs/sling" %>
+
+<sling:defineObjects />
+
+<%
+    String dialogSitewideDataPath = slingRequest.getRequestPathInfo().getSuffix();
+    Session dialogSitewideSession = resourceResolver.adaptTo(Session.class);
+
+    // Ensure the path exists in the JCR so that we don't get a NPE
+    // when attempting to load the dialog.
+    JcrUtil.createPath(dialogSitewideDataPath, "nt:unstructured", "nt:unstructured", dialogSitewideSession, false);
```

Comment:
This is fairly problematic as presumably not all users will have write access to the sitewide data path. Would IMHO be worth investigating the case where write permissions aren't granted.